nixos/tests/plasma5.nix

```diff
@@ -32,34 +20,40 @@ import ./make-test.nix ({ pkgs, ...} :
     };
     hardware.pulseaudio.enable = true; # needed for the factl test, /dev/snd/* exists without them but udev doesn't care then
     virtualisation.memorySize = 1024;
-    environment.systemPackages = [ sddm_theme ];
   };
 
   testScript = { nodes, ... }: let
     user = nodes.machine.config.users.users.alice;
     xdo = "${pkgs.xdotool}/bin/xdotool";
   in ''
-    startAll;
+    start_all()
+
     # wait for log in
-    $machine->waitForFile("/home/alice/.Xauthority");
-    $machine->succeed("xauth merge ~alice/.Xauthority");
+    machine.wait_for_file("${user.home}/.Xauthority")
+    machine.succeed("xauth merge ${user.home}/.Xauthority")
 
-    $machine->waitUntilSucceeds("pgrep plasmashell");
-    $machine->waitForWindow("^Desktop ");
+    # Check plasmashell started
+    machine.wait_until_succeeds("pgrep plasmashell")
+    machine.wait_for_window("^Desktop ")
 
-    # Check that logging in has given the user ownership of devices.
-    $machine->succeed("getfacl -p /dev/snd/timer | grep -q alice");
+    # Check that logging in has given the user ownership of devices
+    machine.succeed("getfacl -p /dev/snd/timer | grep -q ${user.name}")
 
-    $machine->execute("su - alice -c 'DISPLAY=:0.0 dolphin &'");
-    $machine->waitForWindow(" Dolphin");
+    # Run Dolphin
```

Comment:
how about making these comments subtests? That would be nice as it would give this semantic info in the log

Reply:
I thought of this, but I noticed a test passing with `3 of 4 test passed`?

Reply:
how did you group the subtests?

Reply:
ping ;-)

Reply:
Exactly like how the comments are here I think.

Reply:
Ah, turns out that was a bug I noticed https://github.com/NixOS/nixpkgs/commit/ba03a1b731172c975ea9214f8370290d51677e42#commitcomment-36120265. Will switch to this now as it should be working properly.